### PR TITLE
feat: add jsonTrailingCommaFiles

### PR DESCRIPTION
**Pull request by dsherret**
This makes it that someone needs to explicitly opt into special file handling for json as jsonc.

Reasons:

1. Treating .json files (event tsconfig.json) can cause problems in some respositories, leading to them needing to disable trailing commas for even jsonc: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67139
2. Someone may want this behaviour for more than just tsconfig.json (ex. for also tsconfig.base.json).
3. Seems better for people to determine what works for their project or not.

To get the previous config someone can specify something like:

```
  "jsonTrailingCommaFiles": ["tsconfig.json", ".vscode/settings.json"]
```

FYI @Zamiell

### CODE REVIEW

File: deployment/schema.json [hunk 1]
```diff
@@ -121,8 +100,31 @@
         "description": "Never format with trailing commas."
       }]
     },
+    "jsonTrailingCommaDirectories": {
```
Comment (jakebailey) [inline deployment/schema.json:103]: Is this needed when one can write `.vscode/**.json`?

Comment (dsherret) [reply deployment/schema.json:103]: True. I think I'll remove this and just make it so people have to specify explicit files for now. Ex. `.vscode/settings.json` (I don't want to bother adding a glob matcher atm)

Comment (jakebailey) [reply deployment/schema.json:103]: Oh, right, glob matching is in dprint, not the plugin.

File: benches/bench.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  b.iter(|| {
+    let global_config = resolve_global_config(ConfigKeyMap::new(), &Default::default()).config;
+    let config = resolve_config(ConfigKeyMap::new(), &global_config).config;
+    format_text(json_text, &config).unwrap()
+  });
+}
+
+fn get_single_line_800kb_json() -> String {
+  read_to_string("benches/data/single_line_800kb.json").unwrap()
+}
```
Comment (dsherret) [inline benches/bench.rs:26]: Oops. Didn't meant to commit this lol

**Pull request merged**